src/dbnode/client/fetch_tagged_results_accumulator.go

```diff
@@ -302,6 +333,112 @@ func (accum *fetchTaggedResultAccumulator) AsTaggedIDsIterator(
 	return iter, exhaustive, nil
 }
 
+func (accum *fetchTaggedResultAccumulator) AsAggregatedTagsIterator(
+	limit int,
+	pools fetchTaggedPools,
+) (AggregatedTagsIterator, bool, error) {
+	var (
+		iter      = newAggregateTagsIterator(pools)
+		count     = 0
+		moreElems = false
+	)
+	results := aggregateResultsSortedByTag(accum.aggResponses)
+	sort.Sort(results)
+
+	var tempValues []ident.ID
+
+	accum.aggResponses = aggregateResults(results)
+	accum.aggResponses.forEachTag(func(elems aggregateResults, hasMore bool) bool {
+		tagResult := iter.addTag(elems[0].TagName)
```

Comment:
just checking my understanding: we're guaranteed `elems[0]` exists because we'd only call the lambda from the forEachTag implementation if the group was non-empty yea?

Reply:
Correct yeah.

Reply:
I'll add a note.